app/src/main/java/com/matxowy/flashcardsapp/presentation/availablecategories/adapters/AvailableCategoriesAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.matxowy.flashcardsapp.presentation.availablecategories.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.matxowy.flashcardsapp.data.db.entity.Category
+import com.matxowy.flashcardsapp.databinding.AvailableCategoryItemBinding
+
+class AvailableCategoriesAdapter(private val listener: OnDownloadCategoryClickListener) :
+    ListAdapter<Category, AvailableCategoriesAdapter.AvailableCategoriesViewHolder>(DiffCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AvailableCategoriesViewHolder {
+        val binding = AvailableCategoryItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return AvailableCategoriesViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: AvailableCategoriesViewHolder, position: Int) {
+        val currentItem = getItem(position)
+        holder.bind(currentItem)
+    }
+
+    inner class AvailableCategoriesViewHolder(private val binding: AvailableCategoryItemBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(category: Category) {
+            binding.apply {
+                mtvCategoryName.text = category.name
+                cbDownload.setOnCheckedChangeListener { _, isChecked ->
+                    if (isChecked) {
+                        listener.onDownloadCategoryClick(category)
+                        cbDownload.isEnabled = false
```

Comment:
Gdyby tu było dużo kategorii i ta lista by się scrollowała to byłby tu problem bo cbDownload.isEnabled ustawiało by się "losowo" na elementach. Inaczej mówiąc nowe elementy wchodzące na ekran dostawałyby istniejący ViewHolder i może się tak zdarzyć, że taki nowy element dostanie ViewHolder z cbDownload.isEnabled = false mimo, że będzie pobieralny. Takie stany w bind trzeba zawsze ustawiać na podstawie itemu (category) i wtedy trzeba mieć coś w rodzaju cbDownload.isEnabled = category.isDownloaded.not(). Alternatywnie po pobraniu trzeba usuwać kategorię z listy.